### DISCUSSION TITLE
Not able to receive messages from EMQX Data Bridge

### DISCUSSION THREAD

**Discussion by aroranayan** [emqx/emqx#12793] [Q&A] [answered]
I have configured RMQ to work as a MQTT data bridge for EMQX. The connection status is successful but I am still not able to receive messages on RMQ. Below are the EMQX logs:
![image](https://github.com/emqx/emqx/assets/55055374/ddb4b3c0-c757-414a-85db-d3f24e02ddfe)

**Comment by savonarola** [comment]
Hello!
There's nothing wrong with the logs, and the message is sent to the connection by the bridge. 

Could you check the bridge counters? Does success counter increase when you send the message?
Is there an actual subscriber to this topic on the RMQ side?

**Reply by aroranayan** [reply]
Hi @savonarola 
Yes the success counter is increasing. There is no actual subscriber on the RMQ side for this but I am not able see the messages in the Queue via the RMQ Management dashboard.

**Reply by savonarola** [reply]
An increased success counter means the receiving party (RMQ) explicitly reported success.
A normal behaviour for a broker is to drop messages on which there is no subscription.
Could you
* Try the same with an active subscription (on RMQ side) to the topic of the forwarded message?
* Try sending a similar message directly to RMQ's MQTT interface with some direct client, like `mqttx` or `mosquitto_pub`?

**Reply by aroranayan** [reply]
Hi @savonarola 

I did as suggested by you. I am using MQTT Explorer to send messages to RMQ via MQTT protocol and using a NodeJS receiver to see the received messages on the queue. When I am publishing the messages via MQTT Explorer I can see those received on my NodeJS receiver but again not from the EMQX data bridge.

Below are the logs from EMQX:
<img width="1440" alt="image" src="https://github.com/emqx/emqx/assets/55055374/817e5592-8551-4329-a660-3e65c56d3817">

**Reply by savonarola** [reply]
Hello!

The line with `RECV_Data` prints the actual data received by the EMQX-side client from the remote side. it's `<<64, 2, 0, 11>>` bytes, which is a `PUBACK` packet with `RC=0` (success).

Maybe you could share
* Setting of the bridge.
* Some detailed description how you send test message RMQ directly.
* Also description of how the messages are sent to EMQX by a real client.

**Reply by aroranayan** [reply, marked as answer]
Hi @savonarola  Thanks for your inputs. The issue here was both EMQX and RMQ was exposing port 1883 for MQTT. So, when publishing messages via MQTT bridge on port 1833, they were getting delivered to EMQX itself not RMQ. When I updated EMQX config to expose port 1884 instead of 1883, the messages started flowing to RMQ.
